staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas.go

```diff
@@ -61,16 +90,30 @@ func SchemaDeclType(s *schema.Structural, isResourceRoot bool) *DeclType {
 	case "array":
 		if s.Items != nil {
 			itemsType := SchemaDeclType(s.Items, s.Items.XEmbeddedResource)
+			maxItems := int64(noMaxLength)
+			if s.Items.ValueValidation != nil && s.Items.ValueValidation.MaxItems != nil {
+				maxItems = *s.Items.ValueValidation.MaxItems
+			}
+			if maxItems == noMaxLength {
```

Comment:
only estimate if they don't set an explicit MaxItems? if they explicitly set maxItems to MaxInt, we should use their super long value:
```suggestion
			} else {
```

Reply:
Makes sense - fixed